Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2346,89 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-howto-row-factory:
+
+How to create and use row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represents each row as a :class:`tuple`.
+If a :class:`!tuple` does not suit your needs,
+you can use the :class:`sqlite3.Row` class
+or a custom :attr:`~Cursor.row_factory`.
+
+While :attr:`!row_factory` exists as an attribute both on the
+:class:`Cursor` and the :class:`Connection`,
+it is recommended to set :class:`Connection.row_factory`,
+so all cursors created from the connection will use the same row factory.
+
+:class:`!Row` provides indexed and case-insensitive named access to columns,
+with minimal memory overhead and performance impact over a :class:`!tuple`.
+To use :class:`!Row` as a row factory,
+simply assign it to the :attr:`Connection.row_factory` attribute:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+
+Query results are now returned as :class:`!Row` instances:
+
+.. doctest::
+
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0]         # Access by index.
+   'Earth'
+   >>> row["name"]    # Access by name.
+   'Earth'
+   >>> row["RADIUS"]  # Column names are case-insensitive.
+   6378
+
+.. doctest::
```

Comment:
I feel it would be better to draw a line here to distinguish between using existing/default factories, and defining custom ones.  In addition, I don

```suggestion
It is also possible to define a custom :attr:`~Connection.row_factory`
by specifying a :term:`callable` that accepts two arguments,
a :class:`Cursor` object and the :class:`!tuple` of row values,
and returns a custom object representing an SQLite row.

.. doctest::
```

Reply:
Yeah, it _definitely_ needs _something_ here, as its otherwise rather confusing.

However, the information describing the API of the callable accepted should be (and currently is) described in the API reference, rather than here, Therefore, I would instead just introduce the actual example here, i,e, something like:

```suggestion
To create and use a custom :attr:`~Cursor.row_factory`,
in this case returning a :class:`dict` mapping column names to values:

.. doctest::
```

> In addition, I don

Master Qui-Gonn, more to say, have you?

Reply:
Yes, the description of the callable is already given in the reference. Also, a how-to is not a tutorial; let's try to keep it as short and to-the-point as possible.

I prefer CAMs suggestion.